lib/battle_boats/console_ui.rb

```diff
@@ -16,7 +17,12 @@ def greet
     end
 
     def display_board(board)
-      output.puts board_formatter.format_board(board, hide_ships: true)
+      hide_ships = if BattleBoats::DEVELOPMENT
```

Comment:
No offense, but this is pretty ugly :smile:. Is there a way you can confidently ensure that `BattleBoats::DEVELOPMENT` is always a boolean higher up in the call stack? that might mean pulling this logic up higher, doing validation early, or adding an assertion and bailing if the value is not what you expect. 

You might be interested in http://www.confidentruby.com/ for more patterns around this.

Reply:
No offense taken! Thanks for taking a look at this! 🙏 

There is definitely a better way of doing this. I thought of pushing it down the call stack, (in `battle_boats/board_formatter.rb`), where the boolean is actually used, but I can see that such a high-level concept, i.e. what environment are we running in, should be as high up as possible. 

Do we have a copy of _Confident Ruby_? I'd love to read it!